### DISCUSSION TITLE
Erweiterung zum "Energie-Dashboard"

### DISCUSSION THREAD

**Discussion by spirit0607** [solectrus/solectrus#3829] [Ideas]
SOLECTRUS ist aus der Idee entstanden die doch sehr bescheidene SENEC App abzulösen, bzw. zu ergänzen. Seit dem ist sehr sehr viel passiert (Integration Wärmepumpe, separate Verbraucher, und und und.) Der Nutzungsumfang geht im Grunde schon über ein reines Photovoltaik Dashboard hinaus. SOLECTRUS kann nämlich z.B. auch für Leute interessant sein, welche gar keine PV Anlage haben, sondern einfach nur ihren eigenen Stromverbrauch (Hausverbrauch, Wärmepumpe) tracken und nachvollziehen möchten. Für den Bezug des Hausstromverbrauches braucht es im Grunde nur einen WLAN IR Schreib-Lesekopf für den Stromzähler und schon kann es losgehen.

Man könnte nun überlegen, SOLECTRUS vom Photovoltaik Dashboard zum Energie Dashboard zu erweitern, indem man noch den Wasserverbrauch (Ermittelbar z.B. über das Projekt Produkt AI-on-the-edge) https://github.com/jomjol/AI-on-the-edge-device und / oder den Gasverbrauch integriert (z.B. mit einem Tasmota https://community.simon42.com/t/gaszaehler-mit-tasmota-und-mqtt-auslesen-und-in-ha-integrieren/8792

Eine solche Integration kann auch z.B. in HomeAssiatant vorgenommen werden.

Daher wäre aus meiner Sicht auch eine stärke Anbindung an Home Assistant sinnvoll um einen größeren Nutzerkreis zu erlangen. Home Assistant verfügt z.B. auch über eine InfluxDB integration. Somit könnten die Sensoren direkt von HomeAssistant in die InfluxDB integriert werden. Vorteil hier wäre, dass kein MQTT oder Collectoren notwendig wären. Am besten wäre aus meiner Sicht daher eine SOLECTRUS integration im HACS Store, welche bei der Konfiguration fragt, welche Sensoren integriert werden möchten und dann eine passende configuration.yaml für HomeAssistant schreibt und eine passende docker-compose.yml und .env.

SOLECTRUS selber müsste dann am besten noch im Add-On Store von HomeAssistant verfügbar sein, um nutzerfreundlich SOLECTRUS auf dem bestehenden HomeAssistant Rechner zu installieren. Dies würde den Nutzerkreis erheblich erweitern und auch die Konfiguration wäre für viel mehr User leichter durchzuführen.

Vielleicht stelle ich mir das alles viel zu einfach und nicht realisierbar vor. Ist aber einfach nur mal ein Gedankenspiel und Vorschlag. Ich habe leider keinerlei Kenntnisse vom programmieren, daher kann ich da leider nicht mitwirken.

**Comment by ledermann** [comment]
Herzlichen Dank für deine Vorschläge - und Happy New Year :)

Ja, es stimmt, SOLECTRUS ist über die Zeit (mittlerweile vier Jahre!) sehr gewachsen. Und ja, auch ohne Photovoltaik ist eine Nutzung denkbar, indem man sich auf den Stromverbrauch konzentriert, mit den [selbstdefinierten Stromverbrauchern](https://github.com/orgs/solectrus/discussions/3749#discussioncomment-11690923) lässt sich einiges darstellen.

An einem Ausbau der Einsatzszenarien von SOLECTRUS bin ich grundsätzlich interessiert. Hier meine Gedanken im Einzelnen:

- Eine verbesserte Unterstützung für Installationen ohne Photovoltaik kann ich mir sehr gut vorstellen. Einiges ist bereits jetzt möglich, wenn man die Sensoren für die Erzeugung, Einspeisung und Akku einfach leer lässt.

- Was aus meiner persönlichen Sicht besonders interessant ist, wäre eine erweiterte Unterstützung der großen Stromverbraucher E-Auto und Wärmepumpe. Hier fände ich es besonders interessiert, den Input (also den verbrauchten Strom) dem Output (also die erzeugte Wärme bzw. die gefahrenen Kilometer) gegenüber zu stellen. Dazu würde dann auch Effizienz-Kennzahlen gehören, und wie sich diese im Laufe der Zeit verändern. Bei der Wärmepumpe wären das z.B. die Arbeitszahl oder die kWh pro qm Wohnfläche. Beim E-Auto wären das z.B. die kWh oder EUR pro 100 km. Auch die Unterstützung mehrerer E-Autos oder Wallboxen (wie bereits mehrfach gewünscht) wäre interessant.

- Themen wie Gas/Öl und Wasser wären für einzelne User vermutlich interessant, aber da sehe ich ein Motivationsproblem auf meiner Seite: In unserem Haus gibt es keine fossile Heizung mehr. Und der Wasserverbrauch ist bei uns eine Größe, die seit Jahren konstant ist und auch keine nennenswerten Einsparpotentiale bietet. Das wären recht langweilige Auswertungen für mich persönlich. Mein Interesse gilt daher weiterhin eher dem elektrischen Strom.

- SOLECTRUS in den HomeAssistant zu integrieren, ist nicht unbedingt das, was ich mir vorstelle. SOLECTRUS ist als eigenständiges Projekt mit eigener Datenhaltung konzipiert. Ich habe nicht im Detail evaluiert, was eine Integration in den HomeAssistant bedeuten würde, aber ich gehe sehr davon aus, dass das alles andere als trivial ist. Für viele Anwendungsfälle ist HomeAssistant auch gar nicht nötig (z.B. für die SENEC-User, die einzig den SENEC-Collector für die Datensammlung verwenden). Andererseits hat der HomeAssistant eine erheblich größere Community als SOLECTRUS, und es gibt viele User, die HomeAssistant bereits verwenden. Eine Integration könnte also auch neue User für SOLECTRUS bringen. Aber die dafür notwendigen Umbauarbeiten und konzeptionellen Änderungen schrecken mich derzeitig vor einem solchen Schritt ab.

- Die Finanzierung von SOLECTRUS ist weiterhin ein Problem. Die Einnahmen aus den Sponsoring-Abos decken meine Kosten nur zu einem winzigen Bruchteil. Ich zahle jeden Monat einige Tausender drauf, d.h. mein Unternehmen macht entsprechende Verluste. Es gibt zwar etliche sehr nette und engagierte User, die auch bereit sind, für SOLECTRUS den Abopreis von €5 im Monat zu zahlen, aber es sind noch viel zu wenige. Die allermeisten User konnten sich bislang nicht zu einem Abo durchringen.

- Eine Funktionalität zu implementieren, die ich sie selbst nicht nutze, wäre somit realistischerweise nur bei angemessener Vergütung denkbar. Ich habe derartige Einzelaufträge in der Vergangenheit gemacht. Wer also eine Funktionalität wünscht und den Entwicklungsaufwand dafür bezahlen möchte, mit dem komme ich gerne ins Gespräch :-)

**Comment by spirit0607** [comment]
Hallo und auch von mir frohes neues Jahr! :-)

> Die Finanzierung von SOLECTRUS ist weiterhin ein Problem. Die Einnahmen aus den Sponsoring-Abos decken meine Kosten nur zu einem winzigen Bruchteil.

Das glaube ich dir sofort. Das war auch generell mein Ansatz bei dem Vorschlag. Ich könnte mir nämlich vorstellen, dass der Benutzerkreis für die rentable Entwicklung von SOLECTRUS zu klein ist und eine Erweiterung in diesem Rahmen eher schwierig ist. Daher der Gedanke zu HomeAssistant. HomeAssistant ist etabliert und hat einen enorm großen Nutzerkreis. 

> SOLECTRUS ist als eigenständiges Projekt mit eigener Datenhaltung konzipiert. 

Genau das soll SOLECTRUS auch bleiben. Ich hatte mich etwas schlecht ausgedrückt. Also der Vorschlag beruht auf zwei Dingen:

1.) SOLECTRUS als Add-On in den HomeAssistant Store integrieren
Dies soll nur die Installation von SOLECTRUS vereinfachen. Denn seien wir mal ehrlich, wie viele User wissen, wie man einen Docker Container installiert oder mit Docker Compose umgeht. HomeAssistant wird einfach dadurch installiert, dass man sich ein SD Karten Image herunterlädt, dies auf SD Karte flasht und in einen Raspberry Pi steckt. Das bekommen deutlich mehr User hin. SOLECTRUS könnte man auch auf diese Weise zur Verfügung stellen, allerdings würde man dafür dann alleine einen Raspberry Pi benötigen. Zudem bleibt noch das Problem der integration der Sensoren und der zur Verfügung Stellung der Daten. Dies ist der nächste relativ komplizierte Schritt, nämlich über MQTT, denn viele der USER haben vielleicht keinen SENEC Speicher oder nutzen nur Shellys. Daher der zweite Schritt: 

2.) Die KONFIGURATION von SOLECTRUS über eine HomeAssistant Integration:
Die meisten PV Anlagen, Strommessgeräte, usw. haben bereits eine HomeAssistant integration. Viele benutzen diese Geräte auch schon in HomeAssistant. Die Sensoren sind daher auch bereits vorhanden oder können leicht hinzugefügt werden. Nun müssten nun diese Messdaten noch in die InfluxDB von SOLECTRUS. Dies könnte mit HomeAssistant Boardmitteln umgesetzt werden. 
https://www.home-assistant.io/integrations/influxdb/
Diese Integration kann Sensordaten auch in eine externe InfluxDB (nämlich die von SOLECTRUS) schreiben. Der Weg über MQTT könnte man sich daher sparen. 
Die Idee von mir war nun, eine HomeAssitant Integration zu schreiben, welche bei der Konfiguration fragt, welche HomeAssistant Sensoren für SOLECTRUS genutzt werden sollen. Das Programm würde dann die .env und die compose.yml für SOLECTRUS und die configuration.yaml für HomeAssistant erstellen. 

SOLECTRUS könnte auch weiterhin auf einem separaten System betrieben werden, oder eben als Add-on aus dem Add-on Store von HomeAssistant installiert werden
<img width="2709" alt="Bildschirmfoto 2025-01-03 um 10 42 58" src="https://github.com/user-attachments/assets/9c8c4445-632e-4049-bc7d-4d48d4422681" />

Die Integration von Gas und Wasser wäre denn noch ein weiterer optionaler Schritt. Ich selber habe ich keine Gas Heizung mehr und der Wasserverbrauch ist auch konstant, aber ich dachte, dass das Fundament bereits da ist und nur die Messwerte ggf. angepasst werden müssten.

**Reply by ledermann** [reply]
Danke für deinen ausführlichen Input, ich schätze das sehr!

Grundsätzlich ist die Einfachheit der Installation ein äußerst wichtiges Thema. Ich merke durchaus, dass viele User ihre Schwierigkeiten mit Docker haben und es oft an Verständnisproblemen hakt. Wie viele Interessenten da draußen SOLECTRUS gerne installieren würden, aber bereits an der Installation scheitern und für mich nie sichtbar werden, kann ich gar nicht abschätzen.

Die Möglichkeit, über die [InfluxDB-Integration](https://www.home-assistant.io/integrations/influxdb/) Werte direkt von HomeAssistant zu SOLECTRUS zu bringen (also ohne MQTT) hatte ich noch nicht auf dem Schirm. Das dürfte bereits jetzt möglich sein. Wer den MQTT-Collector nur zum Abholen von Werten aus dem HomeAssistant nutzt, kann diese Möglichkeit als Alternative prüfen.

Das andere Thema: Ich bin da zwar nicht tief eingetaucht, aber es könnte tatsächlich machbar sein, ein Add-On für HomeAssistant zu erstellen, sodass SOLECTRUS innerhalb von HA läuft. Das wäre dann optional, d.h. man **könnte** SOLECTRUS als Add-on für HA installieren, kann aber auch die bisherige Installationsweise nutzen.
Es gibt Dokumentation dazu: https://developers.home-assistant.io/docs/add-ons/

Es müsste dann zusätzlich, wie von dir erwähnt, eine Konfiguration erstellt werden, mit der ein User festlegen kann, welche Messwerte aus HomeAssistant in SOLECTRUS verfügbar sein sollen.

Alles in allem wäre das sicherlich ein interessantes Projekt, aber vermutlich auch ein größerer Zeitfresser. Ich habe vor einiger Zeit den [Konfigurator](https://docs.solectrus.de/installation/konfigurator) gebaut, der auf eine andere Weise eine Vereinfachung der Installation zum Ziel hat. Auch das war äußerst aufwändig.

Ansonsten noch: Für HA-Add-Ons gibt es generell eine Bedingung: 

> Add-ons are only available if you've used the Home Assistant Operating System or Home Assistant Supervised [installation](https://www.home-assistant.io/installation) method. If you installed Home Assistant using any other method then you cannot use add-ons. Often you can achieve the same manually, refer to the documentation by the vendor of the application you'd like to install.
https://www.home-assistant.io/addons

Du hast deinen HA vermutlich auf diese Weise installiert. Das führt dann dazu, dass der HA das gesamte Gerät "kapert", es läuft dann das "Home Assistant OS", wenn ich es richtig verstehe. Ich habe das bei mir auf einer Synology NAS in einer virtuellen Maschine mal gemacht - einfach und schön finde ich das aber nicht.

Ein weiterer Punkt ist: Da HomeAssistant kostenfrei ist, könnte es sein, dass kaum jemand bereit ist, für SOLECTRUS als Add-On ein kostenpflichtiges Abo abzuschließen. Zusätzliche User würde also nicht unbedingt zusätzliche Einnahmen bedeuten. Diese Einschätzung ist natürlich aus der Hüfte geschossen, möglicherweise liege ich damit falsch :)

Insgesamt bin ich weiterhin skeptisch, ob sich der zusätzliche Aufwand derzeitig (für mich) lohnt.

**Reply by Zeppelin-PV** [reply]
mit Interesse verfolge ich diesen Dialog zu diesem Thema
ja, auch ich benutze HomeAssistant (HA) und Solectrus
aber - diese zwei Welten miteinander zu verschmelzen wäre "eine Sünde"
alleine schon der Bereich Statistik und Auswertung von Solectrus sucht seinesgleichen
HA hat aber natürlich auch seinen Reiz. Es ist in der Tat so, dass sich hier "blutige Anfänger" relativ schnell zurechtfinden.
Und, es ist äußerst praktisch verschiedene Plattformen zusammenfügen zu können.
Das integrierte Energie Dashboard von HA ist aber weit weg von dem was Solectrus leistet, oder anders ausgedrückt,
hier wird dann ebenfalls viel Hirnschmalz verlangt wenn man darüberhinaus Statistiken und Auswertungen machen möchte.
Solectrus hingegen ist zwar in der Tat bei der Installation und Anpassung etwas schwieriger, dafür muss man sich um den Rest nicht kümmern.
Auch ich bin der Meinung, dass ein Add On oder eine Implementierung in HA Solectrus eventuell bekannter macht, aber kein Benefit
für Georg bringen würde. Weil -?
HA ist bis auf die Anschaffung von Hardware kostenlos. Und nicht nur Schwaben, so wie ich, wissen es zu schätzen alles umsonst
herunterladen zu können. Solectrus brächte eine IIntegration keinerlei finanziellen Vorteil.
Ich sehe übrigens HA als Ersatz einer Modelleisenbahn. Hier kann man schalten und walten wie bei Märklin, Fleischmann & Co
Also, - lassen wir die zwei Welten nebeneinander werkeln - ohne Kernfusion

beinahe vergessen -
dass man vom HA über ioBroker sich die Daten zu Solectrus holen kann für die custom consumer ist doch beispielhaft
dass jede Plattform ihre Daseinsberechtigung hat.
